plugins/database/src/db_plugin.cpp

```diff
@@ -5089,6 +5090,30 @@ irods::error db_reg_zone_op(
         return PASS( ret );
     }
 
+    // =-=-=-=-=-=-=-
+    // validate the connection string is well formed
+    const std::size_t addr_len = std::strlen(_zone_conn_info);
+    if (addr_len == 0) {
+        rodsHostAddr_t addr{};
+        // _zone_conn_info is const, so copy it
+        std::unique_ptr<char[]> addr_buf( // NOLINT(cppcoreguidelines-avoid-c-arrays, modernize-avoid-c-arrays)
+            new char[addr_len + 1]);
+        std::strncpy(addr_buf.get(), _zone_conn_info, addr_len + 1);
+        status = parseHostAddrStr(addr_buf.get(), &addr);
+        if (status < 0) {
+            log_db::info("chlRegZone failed to validate connection info {}", status);
+            return ERROR(status, "failed to validate connection info");
+        }
+        if (std::strlen(addr.hostAddr) == 0) {
+            log_db::info("chlRegZone failed to validate connection info due to empty hostname {}", status);
```

Comment:
`{}` -> `[{}]`

Perhaps we should log the connection string instead of `status`?
If `status` is negative, then the block above will catch it. What does it mean here?

_Same for the other locations._